Stratis.Bitcoin.Cli/Program.cs

```diff
@@ -23,25 +23,40 @@ static void Main(string[] args)
             }
 
             // hack until static flags are removed
+            string blockchain = "bitcoin";
             if (args.Any(a => a.Contains("stratis")))
             {
+                blockchain = "stratis";
                 var s = Network.StratisMain;
                 var st = Network.StratisTest;
             }
+
+            // The first argument is the network name
             var network = Network.GetNetwork(args.First());
-            NodeSettings nodeSettings = NodeSettings.FromArguments(args, network.Name);
+            NodeSettings nodeSettings = NodeSettings.FromArguments(args, blockchain, network);
 
             var rpcSettings = new RpcSettings();
             rpcSettings.Load(nodeSettings);
 
-            // TODO: make rpc port injectable to the RPCClient
-            //var rpcPort = rpcSettings.RPCPort > 0 ? rpcSettings.RPCPort : network.RPCPort;
+            // Find the binding to 127.0.0.1 or the first available. The logic in RPC settings ensures there will be at least 1.
+            System.Net.IPEndPoint nodeEndPoint = rpcSettings.Bind.Where(b => b.Address.ToString() == "127.0.0.1").FirstOrDefault() ?? rpcSettings.Bind[0];
+
+            // Initilize the RPC client with the configured or passed userid, password and endpoint
+            RPCClient rpc = new RPCClient($"{rpcSettings.RpcUser}:{rpcSettings.RpcPassword}", new Uri($"http://{nodeEndPoint}"));
 
-            RPCClient rpc = new RPCClient("user:pass", "http://127.0.0.1", network);
-            
-            var response = rpc.SendCommand(args.ElementAt(1), args.Skip(2));
+            try
+            {
+                // Execute the RPC command
+                RPCResponse response = rpc.SendCommand(args.ElementAt(1), args.Skip(2).ToArray());
 
-            Console.WriteLine(response.ResultString);
+                // Return the result as a string to the console
+                Console.WriteLine(response.ResultString);
```

Comment:
Next iteration of this code, can we use NLog to log things to a file?
Just so that we can have a history of our requests when we close the console.